test/test_collection.py

```diff
@@ -192,17 +192,15 @@ def test_create(self):
             lambda: "create_test_no_wc" not in db.list_collection_names(),
             "drop create_test_no_wc collection",
         )
-        Collection(db, name="create_test_no_wc", create=True)
+        db.create_collection("create_test_no_wc")
```

Comment:
Did we remove support for `Collection(db, name="create_test_no_wc", create=True)`?

Reply:
Yes, we decided to remove that option entirely from the driver.

Reply:
Then this can't be merged in the 4.x branch since it's a major breaking change. So we either need to bring it back or change the plan on when we can release this.

Reply:
What if we make the full release of the async API the 5.0 release?

Reply:
It sounds simpler to add this option back in for the sync api. Isn't that what _IS_SYNC is for?:
```python
def __init__(...):
    ...
    if _IS_SYNC and <should auto create>:
        await self._create()
```

Reply:
Do you have a list of other known breaking changes or is this the only one?

Reply:
I'm not a huge fan of having two arguments (`create` and `session`) for `AsyncCollection` that aren't valid. Doing so requires either keyword arguments everywhere we construct an `AsyncCollection` to keep the existing argument order or just passing `create=False` everywhere. 

My recollection of the original reasoning behind this change is that users shouldn't be using `create=True` in the first place. If that's accurate, I'd much rather remove it and have a breaking change than continue to support something we don't think should be supported.

Reply:
I don't recall any other breaking changes. I would expect our tests to catch all breaking changes for the synchronous API. Is that a fair assumption?